packages/calypso-build/webpack.config.js

```diff
@@ -0,0 +1,119 @@
+/**
+ **** WARNING: No ES6 modules here. Not transpiled! ****
+ */
+/* eslint-disable import/no-nodejs-modules */
+
+/**
+ * External dependencies
+ */
+const path = require( 'path' );
+const webpack = require( 'webpack' );
+const DuplicatePackageCheckerPlugin = require( 'duplicate-package-checker-webpack-plugin' );
+const FileConfig = require( './webpack/file-loader' );
+const Minify = require( './webpack/minify' );
+const SassConfig = require( './webpack/sass' );
+const TranspileConfig = require( './webpack/transpile' );
+const wordpressExternals = require( './webpack/wordpress-externals' );
+
+/**
+ * Internal dependencies
+ */
+// const { workerCount } = require( './webpack.common' ); // todo: shard...
+
+/**
+ * Internal variables
+ */
+const isDevelopment = process.env.NODE_ENV !== 'production';
+
+/**
+ * Return a webpack config object
+ *
+ * Arguments to this function replicate webpack's so this config can be used on the command line,
+ * with individual options overridden by command line args.
+ *
+ * @see {@link https://webpack.js.org/configuration/configuration-types/#exporting-a-function}
+ * @see {@link https://webpack.js.org/api/cli/}
+ *
+ * @param  {object}  env                           environment options
+ * @param  {object}  argv                          options map
+ * @param  {object}  argv.entry                    Entry point(s)
+ * @param  {string}  argv.'output-path'            Output path
+ * @param  {string}  argv.'output-filename'        Output filename pattern
+ * @param  {string}  argv.'output-library-target'  Output library target
+ * @return {object}                                webpack config
+ */
+function getWebpackConfig(
+	env,
+	{
+		entry,
+		'output-path': outputPath = path.join( __dirname, 'dist' ),
+		'output-filename': outputFilename = '[name].js',
+		'output-libary-target': outputLibraryTarget = 'window',
+	}
+) {
+	const workerCount = 1;
+	const cssFilename = '[name].css';
+
+	const webpackConfig = {
+		bail: ! isDevelopment,
+		context: __dirname,
+		entry,
+		mode: isDevelopment ? 'development' : 'production',
+		devtool: process.env.SOURCEMAP || ( isDevelopment ? '#eval' : false ),
+		output: {
+			path: outputPath,
+			filename: outputFilename,
+			libraryTarget: outputLibraryTarget,
+		},
+		optimization: {
+			minimize: ! isDevelopment,
+			minimizer: Minify( {
+				cache: process.env.CIRCLECI
```

Comment:
Would we expect to parametrize all these `process.env` variables?

Reply:
We've been discussing that, e.g. using the function's `env` arg. We decided to merge this for an i1 since it seems to work well enough (see https://github.com/Automattic/jetpack/pull/11802), and tackle those chores later.